client/src/components/chat/MessageContainer.vue

```diff
@@ -1,35 +1,90 @@
 <template>
   <v-container
     class="message-container"
-    v-scroll
-    id="test"
-    ref="test">
-    <Message
-      v-for="(message, index) in messages"
-      :key="index"
-      :message="message"
-      :user="user">
-    </Message>
+    v-if="selectedChannel">
+    <h1>{{ selectedChannel.friendlyName }}</h1>
+    <hr>
+    <div class="messages" ref="messageBody">
+      <Message
+        v-for="(message, index) in messages"
+        :key="index"
+        :message="message"
+        :user="user">
+      </Message>
+    </div>
+
+    <InputContainer
+      @sendMessage="sendMessage">
+    </InputContainer>
   </v-container>
 </template>
 
 <script>
+import InputContainer from '@/components/chat/InputContainer';
 import Message from '@/components/chat/Message';
 
 export default {
     name: 'MessageContainer',
+    components: {
+      InputContainer,
+      Message
+    },
     props: {
-      messages: Array,
+      selectedChannel: Object,
       user: Object,
     },
-    components: {
-      Message
-    }
+    data() {
+      return {
+        messages: undefined
+      }
+    },
+    watch: {
+      selectedChannel() {
+        if (this.selectedChannel) {
+          this.getChannelMessages();
+        }
+      }
+    },
+    methods: {
+      sendMessage(messageText) {
+        this.selectedChannel.sendMessage(messageText)
+          .then(() => {
+            this.getChannelMessages();
+          });
+      },
+      getChannelMessages() {
+        this.selectedChannel.getMessages()
+          .then(messages => {
+            this.messages = messages.items;
+            this.subscribeToNewMessages();
+            this.scrollToBottom();
+          });
+      },
+      subscribeToNewMessages() {
+        this.selectedChannel.on('messageAdded', (message) => {
+          this.messages.push(message);
+        });
+      },
+      scrollToBottom() {
+        let messageBody = this.$refs.messageBody;
+        messageBody.scrollTop = 1000;
```

Comment:
Does 1000 mean scroll to 1000px as the top of the scrolled window? What happens if there are 1000s of messages in the flexbox? and what happens if the person's using a vertical monitor?